cwms-data-api/src/main/java/cwms/cda/data/dao/watersupply/WaterSupplyUtils.java

```diff
@@ -0,0 +1,171 @@
+                .withDisplayValue(lookupType.getDisplayValue())
+                .withTooltip(lookupType.getTooltip())
+                .withActive(lookupType.getActive())
+                .build();
+    }
+
+    public static usace.cwms.db.dao.ifc.cat.LookupType map(LookupType lookupType) {
+        return new usace.cwms.db.dao.ifc.cat.LookupType(lookupType.getOfficeId(),
+                lookupType.getDisplayValue(), lookupType.getTooltip(), lookupType.getActive());
+    }
+
+    public static LocationRefType map(CwmsId projectLocation) {
+        return new LocationRefType(projectLocation.getName(),"",
+                projectLocation.getOfficeId());
+    }
+
+    public static WaterSupplyPump map(LocationType locationType, PumpType pumpType) {
+        if (locationType == null) {
+            return null;
+        }
+        return new WaterSupplyPump.Builder()
+                .withPumpLocation(new Location.Builder(locationType.getLocationRef().getOfficeId(),
+                    locationType.getLocationRef().getBaseLocationId())
+                    .withNearestCity(locationType.getNearestCity())
+                    .withLocationType(locationType.getLocationType())
+                    .withNation(Nation.nationForName(locationType.getNationId()))
+                    .withLatitude(locationType.getLatitude())
+                    .withLongitude(locationType.getLongitude())
+                    .withPublicName(locationType.getPublicName())
+                    .withLongName(locationType.getLongName())
+                    .withDescription(locationType.getDescription())
+                    .withBoundingOfficeId(locationType.getBoundingOfficeId())
+                    .withActive(locationType.getActiveFlag())
+                    .withName(locationType.getLocationRef().getBaseLocationId())
+                    .withStateInitial(locationType.getStateInitial())
+                    .withCountyName(locationType.getCountyName())
+                    .withTimeZoneName(ZoneId.of(locationType.getTimeZoneName()))
+                    .withElevationUnits(locationType.getElevUnitId())
+                    .withElevation(locationType.getElevation())
+                    .withHorizontalDatum(locationType.getHorizontalDatum())
+                    .withVerticalDatum(locationType.getVerticalDatum())
+                    .withLocationKind(locationType.getLocationKindId())
+                    .withMapLabel(locationType.getMapLabel())
+                    .withPublishedLatitude(locationType.getPublishedLatitude())
+                    .withPublishedLongitude(locationType.getPublishedLongitude())
+                    .withOfficeId(locationType.getLocationRef().getOfficeId())
+                    .build())
+                .withPumpType(pumpType).build();
+    }
+
+    public static LocationType map(WaterSupplyPump pump) {
+        return new LocationType(new LocationRefType(pump.getPumpLocation().getName(),
```

Comment:
LocationUtil does the correct type mapping for you already

Reply:
Removed duplicate method